### PR TITLE
Magento2 203 add tests

### PR DESCRIPTION
**Pull request by Skistims**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I updated the CHANGELOG.md

### CODE REVIEW

File: .travis.yml
Comment (augsteyer) [inline .travis.yml]: this is kinda weird, why not add it to composer file?

File: .travis.yml
Comment (augsteyer) [inline .travis.yml]: we should confirm with PO, but I'd dump anything below 5.6

File: CHANGELOG.md
Comment (augsteyer) [inline CHANGELOG.md:15]: what about the other stuff? We don't use this `none` convention. Also, why not outline what the SDK already has right now? Unless it has nothing.

File: composer.json
Comment (augsteyer) [inline composer.json]: not recommended to open it up like that, create a starting version, even if it's just the latest. Why it's not recommended is that your unit tests will not work on lower versions like `4.8` as they are not forward compatible with PSR-4.

File: .travis.yml
Comment (augsteyer) [inline .travis.yml:17]: since you deleted composer-lock file, this actually does nothing, you need to `composer update` instead. 
(why it works currently is because of the require statement above (which I told you to delete :D)

File: tests/Bootstrap.php
Comment (augsteyer) [inline tests/Bootstrap.php]: You should not need to do any of this, you need to namespace the test files properly, e.g. `Shopgate\CloudIntegrationSdk\Tests\Unit\Service\Authenticator`
Your composer file would then need to define where to locate namespaced files like so:
```
"psr-4": {
      "Shopgate\\CloudIntegrationSdk\\": "src/",
      "Shopgate\\CloudIntegrationSdk\\Tests\\": "tests/"
    }
```
Currently your test namespaces are loose and can clash with external modules.

File: tests/Unit/Service/Authenticator/BasicAuthTest.php
Comment (augsteyer) [inline tests/Unit/Service/Authenticator/BasicAuthTest.php]: should not start with a backwards slash in use statements

File: tests/Unit/Service/Authenticator/BasicAuthTest.php
Comment (augsteyer) [inline tests/Unit/Service/Authenticator/BasicAuthTest.php]: should import the class name and use `::class` static  to pull the name instead

File: tests/Unit/Service/Authenticator/BasicAuthTest.php
Comment (augsteyer) [inline tests/Unit/Service/Authenticator/BasicAuthTest.php]: change your inspection to require shorthand array syntax (after you get approval to dump php5.3) + adjust phpcs checker to use "short"

File: tests/Unit/Service/Authenticator/BasicAuthTest.php
Comment (augsteyer) [inline tests/Unit/Service/Authenticator/BasicAuthTest.php:68]: double spacing

File: tests/Unit/Service/Authenticator/BasicAuthTest.php
Comment (augsteyer) [inline tests/Unit/Service/Authenticator/BasicAuthTest.php]: not quite correct, you are getting a mock object. PHPStorm complaining about non-existing methods is correct.

File: tests/Unit/Service/Authenticator/BasicAuthTest.php
Comment (augsteyer) [inline tests/Unit/Service/Authenticator/BasicAuthTest.php]: we should prolly stick to the camelCase naming convention as this standard will not be remembered by most. `testAuthenticateThisThingForSure` is fine

File: tests/Unit/Service/Authenticator/Exception/InvalidAuthenticatorTest.php
Comment (augsteyer) [inline tests/Unit/Service/Authenticator/Exception/InvalidAuthenticatorTest.php]: there is no need to make an alias for this, but if you really want the user to know it's an exception, you can use a better convention of using something like this:
```
use Shopgate\CloudIntegrationSdk\Service\Authenticator\Exception;
...
... = new Exception\InvalidAuthenticator();
```
This allows for adding other exceptions later form that namespace, plus not needing to create extra aliases that need looking up, plus already has that Exception marker that you wanted.

File: .travis.yml
Comment (augsteyer) [inline .travis.yml]: 7.1 should be a string, so wrap with quotes

File: README.md
Comment (augsteyer) [inline README.md:7]: you will need to adjust the example composer require here as the name changed for the repo

File: composer.json
Comment (augsteyer) [inline composer.json]: dont forget to change the name due to repo change! + dont forget to change the composer name as well.
Also remove the shopgate.php file from autoloading (and delete it), we don't need it in the next implementations.

**Review by augsteyer — CHANGES_REQUESTED** [review]
build release package still has the old name for the release zip package. Please rename (dont forget exclude list and gitignore). You also have composer install there, which will not work after my requested changes (I mentioned in other review comment)

File: .travis.yml
Comment (andre-kraus) [reply .travis.yml]: @Stereck approved to dump everything below 5.6.

Comment (andre-kraus) [reply .travis.yml]: But we should also add php 7.2 imho

File: README.md
Comment (andre-kraus) [inline README.md]: The build status badge needs to point to https://api.travis-ci.org/shopgate/connect-integration-sdk-php.svg, also the travis link needs to be updated.

**Review by andre-kraus — CHANGES_REQUESTED** [review]

File: tests/Unit/Service/Authenticator/BasicAuthTest.php
Comment (andre-kraus) [reply tests/Unit/Service/Authenticator/BasicAuthTest.php]: /** @var \Shopgate\CloudIntegrationSdk\ValueObject\Request\Request|\PHPUnit\Framework\MockObject\MockObject $request */

would be more precise, maybe you should add a "use PHPUnit\Framework\MockObject;" to the class to shorten that comment.

**Review by andre-kraus — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (Skistims) [reply CHANGELOG.md:15]: this is just a template for now. I am not preparing for the release, so this is no concern of mine

File: .travis.yml
Comment (Skistims) [reply .travis.yml:17]: I think it will work without the require. It's "install" ?!

File: composer.json
Comment (augsteyer) [reply composer.json]: good, but u still forgot the exclude file list in the deploy folder, gitignore and you did not change this line of code that I was highlighting

File: composer.json
Comment (augsteyer) [inline composer.json:14]: thought we were dumpin' everything below 5.6?

File: composer.json
Comment (augsteyer) [inline composer.json]: should not do this eiether, these versions are mutually exclusive, as I mentioned, we cannot have this bound so low, 4.8 uses PSR-0, while 6+ uses PSR-4. Just use the latest and greatest with whichever you are testing right now. Are you testing with 8? Plop that version in as `^8.0.0` if 7 something, then use that.

File: .gitignore
Comment (augsteyer) [inline .gitignore]: newline

**Review by augsteyer — CHANGES_REQUESTED** [review]

File: composer.json
Comment (Skistims) [reply composer.json:14]: We just dumped the tests, so we don't have to use the "precise" dist from travis anymore.
Techincally, we just dropped 5.3 because of the short array annotation

File: composer.json
Comment (Skistims) [reply composer.json]: I need version 7, because version 8 is not compatible with php 5.6

Comment (augsteyer) [reply composer.json]: just defined `^7.0`, it will prevent 8.0>

**Comment by augsteyer** [comment]
@Skistims seems like the composer is not happy with that phpunit, you will just need to specify them with an OR. Just use the unit 6 version to cover the 5.6 php

File: composer.json
Comment (Skistims) [reply composer.json]: I'll go back to <8
I don't see the advantage of "phpunit/phpunit": "^5.0.0 ||^6.0.0 || ^7.0.0"

**Review by augsteyer — APPROVED** [review]

**Review by andre-kraus — APPROVED** [review]

**Pull request merged**
